compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -2330,10 +2330,16 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                         match (cast_ty_from, cast_ty_to) {
                             (Some(CastTy::Ptr(src)), Some(CastTy::Ptr(dst))) => {
                                 let mut normalize = |t| self.normalize(t, location);
+
+                                // N.B. `struct_tail_with_normalize` only "structurally resolves"
+                                // the type. It is not fully normalized, so we have to normalize it
+                                // afterwards.
```

Comment:
Ah, tricky. Should probably mention this in `struct_tail_with_normalize` docs.

Reply:
Yeah, I'm thinking of better names for this function to make it obvious it's not totally normalizing...